packages/framework/src/types/event.types.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  ITenantDefine,
+  ITriggerPayload,
+  TriggerEventStatusEnum,
+  TriggerRecipientsPayload,
+  TriggerRecipientSubscriber,
+} from '@novu/shared';
+
+export type EventTriggerResult = {
+  /**
+   * Cancel the workflow execution
+   */
+  cancel: () => Promise<CancelEventTriggerResponse>;
+  /**
+   * Response data for the trigger
+   */
+  data: EventTriggerResponse;
+};
+
+export type EventTriggerParams<T_Payload = ITriggerPayload> = {
+  /**
+   * Workflow id
+   */
+  workflowId: string;
+  /**
+   * Recipients to trigger the workflow to
+   */
+  to: TriggerRecipientsPayload;
+  /**
+   * Actor to trigger the workflow from
+   */
+  actor?: TriggerRecipientSubscriber;
```

Comment:
Sounds good, I agree. Will update

Reply:
I have taken this approach and added the `type` import directive for `@novu/shared` rather than declare all the types duplicately in `@novu/framework` as the nested typings are numerous.